src/core/dbus-manager.c

```diff
@@ -1527,8 +1527,10 @@ static void log_caller(sd_bus_message *message, Manager *manager, const char *me
         (void) sd_bus_creds_get_comm(creds, &comm);
         caller = manager_get_unit_by_pid(manager, pid);
 
-        log_info("%s requested from client PID " PID_FMT " ('%s') (from unit '%s')...",
-                 method, pid, strna(comm), strna(caller ? caller->id : NULL));
+        log_info("%s requested from client PID " PID_FMT "%s%s%s%s%s%s...",
+                 method, pid,
+                 comm ? " ('" : "", strempty(comm), comm ? "')" : "",
+                 caller ? " (unit " : "", caller ? caller -> id : NULL, caller ? ")" : "");
```

Comment:
`caller -> id` looks a bit strange with the spaces?

Reply:
Maybe `strjoina()` would make this slightly easier on the eyes?

Reply:
> Maybe `strjoina()` would make this slightly easier on the eyes?

`strjoina()` cannot be used in an argument list… So we'd need to use a temporary variable to assign the result. I don't think that'd be better.

Reply:
> > Maybe `strjoina()` would make this slightly easier on the eyes?
> 
> `strjoina()` cannot be used in an argument list… So we'd need to use a temporary variable to assign the result. I don't think that'd be better.

Ah, I see. In that case it really doesn't make sense.